firmware/lm32/heartbeat.c

```diff
@@ -0,0 +1,83 @@
+
+#include <generated/csr.h>
+#include <generated/mem.h>
+#include <hw/flags.h>
+#include <system.h>
+#include <time.h>
+
+#include "heartbeat.h"
+#include "processor.h"
+#include "hdmi_in0.h"
+#include "hdmi_in1.h"
+#include "pattern.h"
+
+static int heartbeat_status = 0;
+
+void hb_status(int val)
+{
+	if(val==1) { 
+		heartbeat_status = 1; 
+	}
+	else {
+		heartbeat_status = 0;
+	}		
+}
+
+void hb_service(int sink)
+{
+	static int last_event;
+	static int counter;		
+	static bool color_v;
+	if (heartbeat_status==1) {	
+		if(elapsed(&last_event, identifier_frequency_read()/120)) {
```

Comment:
Why are you dividing by 120 here? I assume it is to get a certain frequency? Maybe you should do;

```c
#define HEARTBEAT_FREQUENCY 20 // In Hertz
```